dbms/src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -610,7 +610,7 @@ void registerStorageKafka(StorageFactory & factory)
             skip_broken = static_cast<size_t>(kafka_settings.kafka_skip_broken_messages.value);
         }
 
-        bool intermediate_commit = true;
+        bool intermediate_commit = false;
```

Comment:
Let's put default in KafkaSettings, and move the line 

intermediate_commit = static_cast<bool>(kafka_settings.kafka_commit_every_batch);

here.